pilot/pkg/security/authz/model/generator.go

```diff
@@ -399,6 +399,17 @@ func (g pathGenerator) permission(key, value string, forTCP bool) (*rbacpb.Permi
 		return nil, fmt.Errorf("%q is HTTP only", key)
 	}
 
+	// TODO(jaellio): Should we interpret the path as a PathMatcher if it is an invalid or unsupported by Istio PathTemplate?
```

Comment:
@michaelbannister what are your initial thoughts on this? For my initial draft I didn't add any logic to disallow invalid or unsupported templating. Wanted to get this out for review ASAP.

Reply:
I don't know – is there a general philosophy that Istio applies regarding invalid config? As a user I'd rather be told early that my authorization policy is broken or will never match a path, than spend ages debugging it at runtime, so I'd be in favour of fairly strict validation and failing rather than generating a rule which won't match anything.

But would such an error get surfaced to the client through a validation webhook? I suspect it may depend on how istio's been set up…

Reply:
We should block in validation, and make this an error probably

Reply:
The hard part about error is that previously allowed policies could now be blocked. That may be acceptable though due to a low likelihood?

Reply:
I agree with Keith, I was concerned about disallowing previously allowed paths/policies. However, those paths might technically have been invalid so this might be an acceptable regression.

Reply:
Looks like validation was pushed up @jaellio?

Reply:
I have added basic validation to ensure that if a path contains a support path template (`{*}` or `{**}`) it must also not contain `*` or `**` or any named variable (`{path}`, `{path=*}`, or `{path=**}`).

Reply:
> I don't know – is there a general philosophy that Istio applies regarding invalid config? As a user I'd rather be told early that my authorization policy is broken or will never match a path, than spend ages debugging it at runtime, so I'd be in favour of fairly strict validation and failing with a meaningful error, rather than generating a rule which won't match anything.
> 
> But would such an error get surfaced to the client through a validation webhook? I suspect it may depend on how istio's been set up…

Agree - invalid should be reflected in resource status at least. As we move to CEL validation - not sure if all validations will be possible in CEL and it may be best to use status consistently.

Since this is a very envoy-specific feature - may be worth adopting the same doc and validations that envoy makes. And to confirm - is the syntax in this PR identical to the syntax used by Envoy, or did we make any subtle changes ?